packages/chain-mon/src/wd-mon/service.ts

```diff
@@ -191,16 +191,14 @@ export class WithdrawalMonitor extends BaseServiceV2<Options, Metrics, State> {
         await this.state.messenger.contracts.l2.BedrockMessagePasser.sentMessages(
           proofEvent.args.withdrawalHash
         )
-      const provenAt = `${
-        (dateformat(
-          new Date(
-            (await this.options.l1RpcProvider.getBlock(proofEvent.blockHash))
-              .timestamp * 1000
-          )
+      const provenAt = `${dateformat(
```

Comment:
This is why nested expressions is an antipattern

Reply:
as opposed to something like this?

```
const date = new Date(
  (await this.options.l1RpcProvider.getBlock(proofEvent.blockHash))
    .timestamp * 1000
)
const dateString = dateformat(
  d,
  'mmmm dS, yyyy, h:MM:ss TT',
  true
)
const provenAt = dateString + "UTC"
```
      

My impression was that they are preferred based on the frequency of their occurrence.

Reply:
I recommend a refactor with no nesting of expressions, it makes it much easier to read. Only solidity gets a pass on this because solc sucks and nesting expressions often is an optimization

Reply:
Kelvin dies on a hill re: nested expressions and he is no longer on the eng team day to day

Reply:
lol